test/widgets/subscription_list_test.dart

```diff
@@ -89,18 +89,71 @@ void main() {
     check(isUnpinnedHeaderInTree()).isTrue();
   });
 
-  testWidgets('subscription sort', (tester) async {
+  testWidgets('pinned subscriptions are sorted', (tester) async {
```

Comment:
It'd probably be helpful for the organization of this code (for the reader to see the structure of it) to group all these sorting test cases into a `group` call — like `group('sorting', …`.